### PR TITLE
CLDC-2357 Correct sales bulk upload UPRN errors

### PR DESCRIPTION
**Pull request by natdeanlewissoftwire**
This fixes a number of small issues with errors that appear when UPRN/address data is missing/invalid. Details of the ACs are in the ticket: https://digital.dclg.gov.uk/jira/browse/CLDC-2357

### CODE REVIEW

File: app/models/log.rb [hunk 1]
```diff
@@ -148,13 +148,15 @@ def blank_compound_invalid_non_setup_fields!
...
     if errors.of_kind?(:uprn, :uprn_error)
+      self.uprn = nil
       self.uprn_known = nil
       self.uprn_confirmed = nil
       self.address_line1 = nil
       self.address_line2 = nil
       self.town_or_city = nil
       self.postcode_full = nil
       self.county = nil
+      process_postcode_changes!
```
Comment (natdeanlewissoftwire) [inline app/models/log.rb:158]: necessary to clear la

File: app/mailers/bulk_upload_mailer.rb [hunk 2]
```diff
@@ -5,16 +5,16 @@ class BulkUploadMailer < NotifyMailer
   FAILED_CSV_ERRORS_TEMPLATE_ID = "e27abcd4-5295-48c2-b127-e9ee4b781b75".freeze
   FAILED_FILE_SETUP_ERROR_TEMPLATE_ID = "24c9f4c7-96ad-470a-ba31-eb51b7cbafd9".freeze
   FAILED_SERVICE_ERROR_TEMPLATE_ID = "c3f6288c-7a74-4e77-99ee-6c4a0f6e125a".freeze
-  HOW_FIX_UPLOAD_TEMPLATE_ID = "21a07b26-f625-4846-9f4d-39e30937aa24".freeze
+  HOW_TO_FIX_UPLOAD_TEMPLATE_ID = "21a07b26-f625-4846-9f4d-39e30937aa24".freeze
```
Comment (natdeanlewissoftwire) [inline app/mailers/bulk_upload_mailer.rb]: :)

File: config/locales/en.yml [hunk 3]
```diff
@@ -73,6 +73,12 @@ en:
             choice:
               blank: You must select how would you like to fix errors
               inclusion: You must select one of the following options for how would like to fix errors
+        forms/bulk_upload_sales_resume/fix_choice:
```
Comment (natdeanlewissoftwire) [inline config/locales/en.yml:76]: this copy (and the above for lettings) is pending a slight possible tweak, hence the do not merge label

Comment (natdeanlewissoftwire) [reply config/locales/en.yml:76]: tweak made

**Review by kosiakkatrina — APPROVED** [review]

**Pull request merged**